### PR TITLE
BTRX-441 Add Liquibase support

### PR DESCRIPTION
**Pull request by vvicario**
## Addresses

https://broadinstitute.atlassian.net/browse/BTRX-441

## Changes
* Use Database Migration plugin to  help us manage database changes. The plugin uses the Liquibase library. 
* Remove drizzle driver and go back to mysql in order to integrate database migration. Using Drizzle driver we have some issues, see the following link for more details https://liquibase.jira.com/browse/CORE-2685 

## Testing

---

- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Get a thumb from @rushtong
- [ ] **Submitter**: Merge to develop using github's "Squash and Merge" feature
- [ ] **Submitter**: Delete branch after merge

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -108,7 +108,8 @@ dependencies {
     compile group: 'commons-fileupload', name: 'commons-fileupload', version: '1.3.3'
     compile group: 'joda-time', name: 'joda-time', version: '2.10'
     compile 'org.grails.plugins:quartz:2.0.13'
-
+    compile group: 'mysql', name: 'mysql-connector-java', version: '5.1.36'
```
Comment (rushtong) [inline build.gradle]: I really haven't looked into this, so I don't know what the best approach is, but can we use a newer version of the connector? It looks like the latest supports the version of mysql we have in orsp. See:
* https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-versions.html
* https://mvnrepository.com/artifact/mysql/mysql-connector-java

File: docs/DEVNOTES.md [hunk 2]
```diff
@@ -39,12 +39,15 @@ implementation should be ever get there (react/angular/etc.).
 * Grails 3.3.8: https://grails.org/download.html
 * Gradle 3.5
 * Groovy 2.4.11
+* Liquibase 3.5.5
```
Comment (rushtong) [inline docs/DEVNOTES.md]: I don't think you need this line here. This is meant to be a list of things the developer might need to install.

File: grails-app/migrations/changesets/changelog-1.0.xml [hunk 3]
```diff
@@ -0,0 +1,489 @@
+<?xml version="1.1" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.5.xsd">
+    <changeSet author="vvicario (generated)" id="1543933537042-1">
+        <createTable tableName="amendment">
```
Comment (rushtong) [inline grails-app/migrations/changesets/changelog-1.0.xml]: We don't need this table.

File: grails-app/migrations/changesets/changelog-1.0.xml [hunk 4]
```diff
@@ -0,0 +1,489 @@
...
+            <column name="project_key" type="VARCHAR(255)">
+                <constraints nullable="false"/>
+            </column>
+            <column name="amendment_number" type="INT">
+                <constraints nullable="false"/>
+            </column>
+        </createTable>
+    </changeSet>
+    <changeSet author="vvicario (generated)" id="1543933537042-2">
+        <createTable tableName="amendment_storage_document">
```
Comment (rushtong) [inline grails-app/migrations/changesets/changelog-1.0.xml]: We don't need this table

File: grails-app/migrations/changesets/changelog-1.0.xml [hunk 5]
```diff
@@ -0,0 +1,489 @@
+<?xml version="1.1" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.5.xsd">
+    <changeSet author="vvicario (generated)" id="1543933537042-1">
```
Comment (rushtong) [inline grails-app/migrations/changesets/changelog-1.0.xml]: nit, can you replace `vvicario (generated)` with just your username, `vvicario`, here and throughout? That would be a lot cleaner.

File: src/main/config/application.yml.ctmpl [hunk 6]
```diff
@@ -5,7 +5,10 @@ hibernate:
     queries: false
     use_second_level_cache: false
     use_query_cache: false
-
+grails.plugin.databasemigration.changelogLocation: grails-app/migrations
+grails.plugin.databasemigration.changelogFileName: changelog-master.xml
+grails.plugin.databasemigration.updateOnStart: true
+grails.plugin.databasemigration.updateOnStartFileName: changelog-master.xml
```
Comment (rushtong) [inline src/main/config/application.yml.ctmpl]: This is yaml, so should be more consistent with the rest of the structure:
```suggestion
grails.plugin.databasemigration:
    changelogLocation: grails-app/migrations
    changelogFileName: changelog-master.xml
    updateOnStart: true
    updateOnStartFileName: changelog-master.xml
```
Also, if you can move it to where the other `grails` properties are, that will also be cleaner.

**Comment by vvicario** [comment]
@rushtong  It's ready for re-review!

**Comment by rushtong** [comment]
@vvicario FYI - Running grails `dbm-changelog-sync` from my machine:

```
java.sql.SQLException: The server timezone value 'UTC' is unrecognized or represents more than one timezone. You must configure either the server or JDBC driver (via the serverTimezone configuration property) to use a more specifc timezone value if you want to utilize timezone support.
```

Needed to add this to the jdbc url to get liquibase to work:

```
&useJDBCCompliantTimezoneShift=true&useLegacyDatetimeCode=false&serverTimezone=UTC
```

That starts up the app which is failing on something unrelated, but the `DATABASECHANGELOG` does end up populated correctly!

**Comment by rushtong** [comment]
I was able to add a new changelog file that was executed on startup (with the above jdbc url change) so this is now functionally validated.

File: src/main/config/application.yml.ctmpl [hunk 7]
```diff
@@ -172,10 +171,10 @@ environments:
         statusUrl: http://bsp/BSP/
     dataSource:
       pooled: true
-      driverClassName: org.drizzle.jdbc.DrizzleDriver
+      driverClassName: com.mysql.jdbc.Driver
       username: {{$orsp.Data.prod_ds_username}}
       password: {{$orsp.Data.prod_ds_password}}
-      url: jdbc:mysql:thin://{{$orsp.Data.prod_ds_host}}/{{$orsp.Data.prod_ds_db}}
+      url: jdbc:mysql://{{$orsp.Data.prod_ds_host}}/{{$orsp.Data.prod_ds_db}}
```
Comment (rushtong) [inline src/main/config/application.yml.ctmpl]: This needs to be:
```suggestion
      url: jdbc:mysql://{{$orsp.Data.prod_ds_host}}/{{$orsp.Data.prod_ds_db}}?useUnicode=true&useJDBCCompliantTimezoneShift=true&useLegacyDatetimeCode=false&serverTimezone=UTC
```

Comment (vvicario) [reply src/main/config/application.yml.ctmpl]: Thank you Greg!

**Comment by rushtong** [comment]
This didn't get merged correctly. Re-opening against master.

**Pull request merged**